plasma_framework/contracts/src/exits/payment/controllers/PaymentPiggybackInFlightExit.sol

```diff
@@ -27,6 +27,9 @@ library PaymentPiggybackInFlightExit {
     uint8 constant public MAX_INPUT_NUM = 4;
     uint8 constant public MAX_OUTPUT_NUM = 4;
 
+    // we use a single exit processing queue
```

Comment:
I don't like that, but since we didn't decide to implement transparent tokens in outputs we don't have a way to enqueue exit to the right queue. That's the reason we put everything into a single queue (no matter what the vault id is). That's problematic, since after we deploy a tx format and exit game that enqueues in the right way (using a proper vault_id), we'll have a discrepancy (some exits will be enqueued with vault_id 1, even though non-id 1 vaults fund that).

Reply:
I would inject the right vault id of `EthVault` and `Erc20Vault` and remove the injection of address and fetch on construction instead.